### ISSUE TITLE
`nested_in()` + `repeated()` produce incorrect error spans.

### ISSUE THREAD

**Issue by ktsimpso** [zesterer/chumsky#577]
Using `1.0.0.alpha6` using a combination of `nested_in` + `repeated()` if there is a parse error in the inner parser, the error is reported with a span relative to the start of the chunk that the outer parser parsed rather than the start of the entire input. (This makes sense from the inner parser's perspective as it does not know about the other input). Though my expectation would be for the combinator to reframe error spans to be consistent with the outer span context.

Example:

```rs
let blank_line = text::newline::<&str, extra::Err<Rich<'_, char>>>()
    .repeated()
    .exactly(2)
    .ignored();
let chunker = any()
    .and_is(blank_line.not())
    .repeated()
    .at_least(1)
    .to_slice()
    .then_ignore(blank_line.or(end()));
let numbers = text::int(10)
    .separated_by(text::newline())
    .allow_trailing()
    .collect::<Vec<_>>();
let parser = numbers.nested_in(chunker).repeated().collect::<Vec<_>>();

let result = parser.parse("1\n2\n\n3\n4\n\n").into_result();
assert_eq!(result, Ok(vec![vec!["1", "2"], vec!["3", "4"]]));

let result = parser.parse("1\na\n\n3\n4\n\n").into_result();
assert_eq!(
    result.map_err(|e| *e.first().unwrap().span()),
    Err(SimpleSpan::new(2usize, 3))
);

let result = parser.parse("1\n2\n\n3\na\n\n").into_result();
assert_eq!(
    result.map_err(|e| *e.first().unwrap().span()),
    Err(SimpleSpan::new(7usize, 8))
); // fails with left: Err(2..3)
 ```

Alternatively is there a nice way for me to manually map the span to the correct outer context?

**Comment by zesterer** [comment]
Yes, you could use `Input::spanned`, like [this example](https://github.com/zesterer/chumsky/blob/main/examples/nested_spans.rs#L19-L21).

That aside, it's not clear to me why `nested_in` is being used here: it's designed primarily for recursive inputs like token trees, not for performing extra validation upon an existing output.

**Comment by ktsimpso** [comment]
> ```rs .then_ignore(text::newline().repeated());```
This works and passes all my tests. Thanks!

@zesterer I still think the default nested_in error span behavior is a bit surprising and does not seem consistent with the other combinators. At the very least I think this should be called out in the docs, though I think ideally the combinator would internally track the proper spans. I'll leave it to you if you want to close this issue or not.
